grafana-plugin/src/components/PluginBridge/PluginBridge.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { FC, ReactElement } from 'react';
+
+import { useAsync } from 'react-use';
+
+import { getPluginSettings } from './PluginService';
+
+export enum SupportedPlugin {
+  Incident = 'grafana-incident-app',
+  MachineLearning = 'grafana-ml-app',
+}
+
+export type PluginID = SupportedPlugin | string;
+
+export interface PluginBridgeProps {
+  plugin: PluginID;
+  // shows an optional component when the plugin is not installed
+  notInstalledComponent?: ReactElement;
+  // shows an optional component when we're checking if the plugin is installed
+  loadingComponent?: ReactElement;
+  children?: ReactElement;
+}
+
+export const PluginBridge: FC<PluginBridgeProps> = ({ children, plugin }) => {
+  const { loading, error, value } = useAsync(() => getPluginSettings(plugin, { showErrorAlert: false }));
```

Comment:
We should probably use this `useAsync` hook more :+1: I didn't even know up to this point we have this in the project :open_mouth: